### ISSUE TITLE
`sanitizeRenderedMarkdown` blocks the renderer 

### ISSUE THREAD

**Issue by chetanrchetu** [microsoft/vscode#204234]
Type: <b>Performance Issue</b>

So when i am using the vs code it is gettin slower the code gets delayed while i am typing

VS Code version: Code 1.86.0 (05047486b6df5eb8d44b2ecd70ea3bdf775fd937, 2024-01-31T10:28:19.990Z)
OS version: Windows_NT x64 10.0.22621
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i5-1035G1 CPU @ 1.00GHz (8 x 1190)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: enabled_on<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_graphite: disabled_off<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled|
|Load (avg)|undefined|
|Memory (System)|7.77GB (1.08GB free)|
|Process Argv|--crash-reporter-id 3e9864da-ab68-48ef-9006-85a7d4bf5c91|
|Screen Reader|no|
|VM|40%|
</details><details>
<summary>Process Info</summary>

```
CPU %	Mem MB	   PID	Process
    0	   124	  1472	code main
    0	   100	  4384	ptyHost
    0	     7	  2840	     conpty-agent
    0	    86	  4372	     C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe
    0	    78	 11568	     C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe
    0	     7	 14952	     conpty-agent
    0	    94	  4500	shared-process
    0	    26	  6228	   crashpad-handler
    0	    94	  7084	fileWatcher [1]
    0	   432	  9636	window [1] (code.ipynb - space_titanic - Visual Studio Code)
    0	    37	  9996	   utility-network-service
    0	   193	 10040	   window
    0	   103	 11376	window [2] (Issue Reporter)
    0	   172	 12700	extensionHost [1]
    0	   245	  1240	     electron-nodejs (JKRwrFlIV.js )
    0	     9	 15384	       C:\WINDOWS\system32\conhost.exe 0x4
    0	   299	  8320	     electron-nodejs (bundle.js )
    0	    16	  9728	     d:\anaconda\envs\machine_learning\python.exe c:\Users\lenovo\.vscode\extensions\ms-toolsai.jupyter-2024.1.0-win32-x64\pythonFiles\vscode_datascience_helpers\kernel_interrupt_daemon.py --ppid 12700
    0	    11	  9408	       C:\WINDOWS\system32\conhost.exe 0x4
    0	    95	 10028	     "D:\Microsoft VS Code\Code.exe" "d:\Microsoft VS Code\resources\app\extensions\markdown-language-features\server\dist\node\workerMain" --node-ipc --clientProcessId=12700
    0	    33	 17236	     d:\anaconda\envs\machine_learning\python.exe c:\Users\lenovo\.vscode\extensions\ms-python.isort-2023.10.1\bundled\tool\lsp_server.py
    0	    10	  9476	       C:\WINDOWS\system32\conhost.exe 0x4
    0	   190	 15716	   gpu-process
```

</details>
<details>
<summary>Workspace Info</summary>

```
|  Window (code.ipynb - space_titanic - Visual Studio Code)
|    Folder (space_titanic): 8 files
|      File types: csv(3) ipynb(1) txt(1)
|      Conf files:;
```

</details>
<details><summary>Extensions (19)</summary>

Extension|Author (truncated)|Version
---|---|---
insert-unicode|bru|0.15.1
doxdocgen|csc|1.4.0
prettier-vscode|esb|10.1.0
code-runner|for|0.12.1
vscode-drawio|hed|1.6.6
better-cpp-syntax|jef|1.17.2
rainbow-csv|mec|3.11.0
vscode-html-format|moh|0.1.6
isort|ms-|2023.10.1
python|ms-|2023.22.1
cmake-tools|ms-|1.16.32
cpptools|ms-|1.19.2
cpptools-extension-pack|ms-|1.3.0
excalidraw-editor|pom|3.7.1
pdf|tom|1.2.2
cmake|twx|0.0.17
vscode-java-debug|vsc|0.56.0
vscode-java-dependency|vsc|0.23.4
vscode-maven|vsc|0.44.0

(2 theme extensions excluded)

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vspor879:30202332
vspor708:30202333
vspor363:30204092
vscod805cf:30301675
binariesv615:30325510
vsaa593cf:30376535
py29gd2263:30899288
c4g48928:30535728
azure-dev_surveyone:30548225
vscrp:30673768
a9j8j154:30646983
962ge761:30951796
pythongtdpath:30769146
welcomedialogc:30910334
pythonidxpt:30866567
pythonnoceb:30805159
asynctok:30898717
pythontestfixt:30902429
pythonregdiag2:30936856
pyreplss1:30897532
pythonmypyd1:30879173
pythoncet0:30885854
pythontbext0:30879054
dsvsc016:30899300
dsvsc017:30899301
dsvsc018:30899302
dsvsc019b:30953937
3ef8e399:30949928

```

</details>

**Comment by rmbennet** [comment]
Same issue.  I am on M1 Pro MacBook Pro with Mac OS 13 Ventura.

I thought it was my computer, but it's only one year old and it only happens when I'm in VS code.  I actually went online to see if I was updated to the latest version.  And noting that the latest version just came out and I am running it (it must have updated automatically), the slowness has to be related to the upgrade.

It's quite frustrating.  It happens When I am typing in code cells, or markdown cells.

**Comment by jsecrest** [comment]
Same issue. Windows 11, Remote development in docker container (image: FROM python:3.12)
I've attached a developer tools performance profile

Type: <b>Performance Issue</b>

Over the last few days I've been experiencing freezing periodically when typing.

VS Code version: Code 1.86.0 (05047486b6df5eb8d44b2ecd70ea3bdf775fd937, 2024-01-31T10:28:19.990Z)
OS version: Windows_NT x64 10.0.22621
Modes:
Remote OS version: Linux x64 5.10.102.1-microsoft-standard-WSL2

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|11th Gen Intel(R) Core(TM) i7-1185G7 @ 3.00GHz (8 x 2995)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: enabled_on<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_graphite: disabled_off<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled|
|Load (avg)|undefined|
|Memory (System)|15.59GB (2.26GB free)|
|Process Argv|--crash-reporter-id bb872725-c55f-42c3-8230-67848914cce6|
|Screen Reader|yes|
|VM|0%|

|Item|Value|
|---|---|
|Remote|Dev Container: Existing Dockerfile|
|OS|Linux x64 5.10.102.1-microsoft-standard-WSL2|
|CPUs|11th Gen Intel(R) Core(TM) i7-1185G7 @ 3.00GHz (8 x 2995)|
|Memory (System)|7.56GB (4.50GB free)|
|VM|0%|
</details><details>
<summary>Process Info</summary>

```
CPU %	Mem MB	   PID	Process
    0	   184	 25616	code main
    0	   112	  4260	window [2] (Issue Reporter)
    0	   556	  4316	window [1] (● 01b-DistributionReview.ipynb - D195-Capstone-ResilienceVsCovid [Dev Container: Existing Dockerfile] - Visual Studio Code)
    0	   140	  5760	   window
    0	   103	 14196	shared-process
    0	   260	 15452	   gpu-process
    0	    24	 18028	   crashpad-handler
    0	   160	 18632	extensionHost [1]
    0	    22	  2432	     "C:\Program Files\Docker\Docker\resources\bin\docker.exe" exec -i -u root -e VSCODE_REMOTE_CONTAINERS_SESSION=4512fbc2-3e56-4f76-bd4e-9b22d6a079321707189593970 a51a562a6d2300e743d12be3fd812275da0d8e4db005424f6d2539e6a1ef91da /root/.vscode-server/bin/05047486b6df5eb8d44b2ecd70ea3bdf775fd937/node -e "
				const net = require('net');
				const fs = require('fs');
				process.stdin.pause();
				const client = net.createConnection({ host: '127.0.0.1', port: 37343 }, () => {
					console.error('Connection established');
					client.pipe(process.stdout);
					process.stdin.pipe(client);
				});
				client.on('close', function (hadError) {
					console.error(hadError ? 'Remote close with error' : 'Remote close');
					process.exit(hadError ? 1 : 0);
				});
				client.on('error', function (err) {
					process.stderr.write(err && (err.stack || err.message) || String(err));
				});
				process.stdin.on('close', function (hadError) {
					console.error(hadError ? 'Remote stdin close with error' : 'Remote stdin close');
					process.exit(hadError ? 1 : 0);
				});
				process.on('uncaughtException', function (err) {
					fs.writeSync(process.stderr.fd, `Uncaught Exception: ${String(err && (err.stack || err.message) || err)}\n`);
				});
			"
    0	     6	 11768	       C:\WINDOWS\system32\conhost.exe 0x4
    0	    15	 14356	       "C:\Program Files\Docker\Docker\resources\bin\com.docker.cli.exe" exec -i -u root -e VSCODE_REMOTE_CONTAINERS_SESSION=4512fbc2-3e56-4f76-bd4e-9b22d6a079321707189593970 a51a562a6d2300e743d12be3fd812275da0d8e4db005424f6d2539e6a1ef91da /root/.vscode-server/bin/05047486b6df5eb8d44b2ecd70ea3bdf775fd937/node -e "
				const net = require('net');
				const fs = require('fs');
				process.stdin.pause();
				const client = net.createConnection({ host: '127.0.0.1', port: 37343 }, () => {
					console.error('Connection established');
					client.pipe(process.stdout);
					process.stdin.pipe(client);
				});
				client.on('close', function (hadError) {
					console.error(hadError ? 'Remote close with error' : 'Remote close');
					process.exit(hadError ? 1 : 0);
				});
				client.on('error', function (err) {
					process.stderr.write(err && (err.stack || err.message) || String(err));
				});
				process.stdin.on('close', function (hadError) {
					console.error(hadError ? 'Remote stdin close with error' : 'Remote stdin close');
					process.exit(hadError ? 1 : 0);
				});
				process.on('uncaughtException', function (err) {
					fs.writeSync(process.stderr.fd, `Uncaught Exception: ${String(err && (err.stack || err.message) || err)}\n`);
				});
			"
    0	    22	  2856	     "C:\Program Files\Docker\Docker\resources\bin\docker.exe" exec -i -u root -e VSCODE_REMOTE_CONTAINERS_SESSION=4512fbc2-3e56-4f76-bd4e-9b22d6a079321707189593970 a51a562a6d2300e743d12be3fd812275da0d8e4db005424f6d2539e6a1ef91da /root/.vscode-server/bin/05047486b6df5eb8d44b2ecd70ea3bdf775fd937/node -e "
				const net = require('net');
				const fs = require('fs');
				process.stdin.pause();
				const client = net.createConnection({ host: '127.0.0.1', port: 37343 }, () => {
					console.error('Connection established');
					client.pipe(process.stdout);
					process.stdin.pipe(client);
				});
				client.on('close', function (hadError) {
					console.error(hadError ? 'Remote close with error' : 'Remote close');
					process.exit(hadError ? 1 : 0);
				});
				client.on('error', function (err) {
					process.stderr.write(err && (err.stack || err.message) || String(err));
				});
				process.stdin.on('close', function (hadError) {
					console.error(hadError ? 'Remote stdin close with error' : 'Remote stdin close');
					process.exit(hadError ? 1 : 0);
				});
				process.on('uncaughtException', function (err) {
					fs.writeSync(process.stderr.fd, `Uncaught Exception: ${String(err && (err.stack || err.message) || err)}\n`);
				});
			"
    0	     6	 11668	       C:\WINDOWS\system32\conhost.exe 0x4
    0	    15	 14060	       "C:\Program Files\Docker\Docker\resources\bin\com.docker.cli.exe" exec -i -u root -e VSCODE_REMOTE_CONTAINERS_SESSION=4512fbc2-3e56-4f76-bd4e-9b22d6a079321707189593970 a51a562a6d2300e743d12be3fd812275da0d8e4db005424f6d2539e6a1ef91da /root/.vscode-server/bin/05047486b6df5eb8d44b2ecd70ea3bdf775fd937/node -e "
				const net = require('net');
				const fs = require('fs');
				process.stdin.pause();
				const client = net.createConnection({ host: '127.0.0.1', port: 37343 }, () => {
					console.error('Connection established');
					client.pipe(process.stdout);
					process.stdin.pipe(client);
				});
				client.on('close', function (hadError) {
					console.error(hadError ? 'Remote close with error' : 'Remote close');
					process.exit(hadError ? 1 : 0);
				});
				client.on('error', function (err) {
					process.stderr.write(err && (err.stack || err.message) || String(err));
				});
				process.stdin.on('close', function (hadError) {
					console.error(hadError ? 'Remote stdin close with error' : 'Remote stdin close');
					process.exit(hadError ? 1 : 0);
				});
				process.on('uncaughtException', function (err) {
					fs.writeSync(process.stderr.fd, `Uncaught Exception: ${String(err && (err.stack || err.message) || err)}\n`);
				});
			"
    0	    22	 13856	     "C:\Program Files\Docker\Docker\resources\bin\docker.exe" exec -i -u root -e VSCODE_REMOTE_CONTAINERS_SESSION=4512fbc2-3e56-4f76-bd4e-9b22d6a079321707189593970 a51a562a6d2300e743d12be3fd812275da0d8e4db005424f6d2539e6a1ef91da /bin/sh
    0	    15	 14472	       "C:\Program Files\Docker\Docker\resources\bin\com.docker.cli.exe" exec -i -u root -e VSCODE_REMOTE_CONTAINERS_SESSION=4512fbc2-3e56-4f76-bd4e-9b22d6a079321707189593970 a51a562a6d2300e743d12be3fd812275da0d8e4db005424f6d2539e6a1ef91da /bin/sh
    0	     6	 28620	       C:\WINDOWS\system32\conhost.exe 0x4
    0	    22	 20840	     "C:\Program Files\Docker\Docker\resources\bin\docker.exe" exec -i -u root a51a562a6d2300e743d12be3fd812275da0d8e4db005424f6d2539e6a1ef91da /bin/sh
    0	     6	 14308	       C:\WINDOWS\system32\conhost.exe 0x4
    0	    15	 27240	       "C:\Program Files\Docker\Docker\resources\bin\com.docker.cli.exe" exec -i -u root a51a562a6d2300e743d12be3fd812275da0d8e4db005424f6d2539e6a1ef91da /bin/sh
    0	    73	 26376	     "C:\Program Files\Microsoft VS Code\Code.exe" c:\Users\Jason\.vscode\extensions\ms-vscode-remote.remote-containers-0.338.1\dist\shutdown\shutdownMonitorProcess \\.\pipe\vscode-remote-containers-064af131-8f2d-4748-88e2-2335cdba0ffa-sock singleContainer Debug c:\Users\Jason\AppData\Roaming\Code\logs\20240205T201953\window1\exthost\ms-vscode-remote.remote-containers 1707189594640
    0	    22	 29236	     "C:\Program Files\Docker\Docker\resources\bin\docker.exe" exec -i -u root a51a562a6d2300e743d12be3fd812275da0d8e4db005424f6d2539e6a1ef91da /bin/sh -c "echo \"New container started. Keep-alive process started.\" ; export VSCODE_REMOTE_CONTAINERS_SESSION=4512fbc2-3e56-4f76-bd4e-9b22d6a079321707189593970 ; /bin/sh"
    0	    15	  4016	       "C:\Program Files\Docker\Docker\resources\bin\com.docker.cli.exe" exec -i -u root a51a562a6d2300e743d12be3fd812275da0d8e4db005424f6d2539e6a1ef91da /bin/sh -c "echo \"New container started. Keep-alive process started.\" ; export VSCODE_REMOTE_CONTAINERS_SESSION=4512fbc2-3e56-4f76-bd4e-9b22d6a079321707189593970 ; /bin/sh"
    0	     6	 28116	       C:\WINDOWS\system32\conhost.exe 0x4
    0	    22	 30080	     "C:\Program Files\Docker\Docker\resources\bin\docker.exe" exec -i -u root -e SHELL=/bin/bash -e VSCODE_AGENT_FOLDER=/root/.vscode-server -w /root/.vscode-server/bin/05047486b6df5eb8d44b2ecd70ea3bdf775fd937 a51a562a6d2300e743d12be3fd812275da0d8e4db005424f6d2539e6a1ef91da /root/.vscode-server/bin/05047486b6df5eb8d44b2ecd70ea3bdf775fd937/bin/code-server --log debug --force-disable-user-env --server-data-dir /root/.vscode-server --use-host-proxy --telemetry-level all --accept-server-license-terms --host 127.0.0.1 --port 0 --connection-token-file /root/.vscode-server/data/Machine/.connection-token-05047486b6df5eb8d44b2ecd70ea3bdf775fd937 --extensions-download-dir /root/.vscode-server/extensionsCache --start-server --disable-websocket-compression --skip-requirements-check
    0	     6	  1784	       C:\WINDOWS\system32\conhost.exe 0x4
    0	    15	 17532	       "C:\Program Files\Docker\Docker\resources\bin\com.docker.cli.exe" exec -i -u root -e SHELL=/bin/bash -e VSCODE_AGENT_FOLDER=/root/.vscode-server -w /root/.vscode-server/bin/05047486b6df5eb8d44b2ecd70ea3bdf775fd937 a51a562a6d2300e743d12be3fd812275da0d8e4db005424f6d2539e6a1ef91da /root/.vscode-server/bin/05047486b6df5eb8d44b2ecd70ea3bdf775fd937/bin/code-server --log debug --force-disable-user-env --server-data-dir /root/.vscode-server --use-host-proxy --telemetry-level all --accept-server-license-terms --host 127.0.0.1 --port 0 --connection-token-file /root/.vscode-server/data/Machine/.connection-token-05047486b6df5eb8d44b2ecd70ea3bdf775fd937 --extensions-download-dir /root/.vscode-server/extensionsCache --start-server --disable-websocket-compression --skip-requirements-check
    0	    70	 22616	fileWatcher [1]
    0	   673	 23764	   window
    0	    38	 24816	   utility-network-service
    0	   101	 26536	   utility-process

Remote: Dev Container: Existing Dockerfile
CPU %	Mem MB	   PID	Process
    0	     0	   189	remote agent
    0	     0	   258	   ptyHost
    0	     0	   290	   fileWatcher
    0	     0	   304	   extension-host
    0	     0	   400	     /usr/local/bin/python /root/.vscode-server/extensions/ms-python.isort-2023.10.1/bundled/tool/lsp_server.py
    0	     0	   411	     /vscode/vscode-server/bin/linux-x64/05047486b6df5eb8d44b2ecd70ea3bdf775fd937/node /vscode/vscode-server/bin/linux-x64/05047486b6df5eb8d44b2ecd70ea3bdf775fd937/extensions/json-language-features/server/dist/node/jsonServerMain --node-ipc --clientProcessId=304
    0	     0	   447	     /vscode/vscode-server/bin/linux-x64/05047486b6df5eb8d44b2ecd70ea3bdf775fd937/node /root/.vscode-server/extensions/ms-python.vscode-pylance-2023.12.1/dist/server.bundle.js --cancellationReceive=file:d3a4d62109c4b1993c1838e96e07687610b42dfe39 --node-ipc --clientProcessId=304
    0	     0	   991	     /vscode/vscode-server/bin/linux-x64/05047486b6df5eb8d44b2ecd70ea3bdf775fd937/node /vscode/vscode-server/bin/linux-x64/05047486b6df5eb8d44b2ecd70ea3bdf775fd937/extensions/markdown-language-features/server/dist/node/workerMain --node-ipc --clientProcessId=304
    0	     0	 36577	     electron-nodejs (jwezhQPZz.js )
    0	     0	 43476	   /bin/sh -c /usr/bin/ps -ax -o pid=,ppid=,pcpu=,pmem=,command=
    0	     0	 43477	     /usr/bin/ps -ax -o pid=,ppid=,pcpu=,pmem=,command=
```

</details>
<details>
<summary>Workspace Info</summary>

```

|  Remote: Dev Container: Existing Dockerfile|    Folder (D195-Capstone-ResilienceVsCovid): 49 files|      File types: csv(10) ipynb(6) md(6) pdf(4) xlsx(3) json(2) html(2)
|                  dockerignore(1) gitattributes(1) gitignore(1)
|      Conf files: devcontainer.json(1) settings.json(1) dockerfile(1);
```

</details>
<details><summary>Extensions (18)</summary>

Extension|Author (truncated)|Version
---|---|---
codespaces|Git|1.16.9
jupyter-keymap|ms-|1.1.2
remote-containers|ms-|0.338.1
remote-wsl|ms-|0.86.0
copilot|Git|1.156.0
copilot-chat|Git|0.12.0
latex-workshop|Jam|9.18.0
rainbow-csv|mec|3.11.0
vscode-docker|ms-|1.28.0
debugpy|ms-|2024.0.0
isort|ms-|2023.10.1
python|ms-|2024.0.0
vscode-pylance|ms-|2023.12.1
jupyter|ms-|2024.1.0
jupyter-keymap|ms-|1.1.2
jupyter-renderers|ms-|1.0.17
vscode-jupyter-cell-tags|ms-|0.1.8
vscode-jupyter-slideshow|ms-|0.1.5


</details><details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vspor879:30202332
vspor708:30202333
vspor363:30204092
vstes627:30244334
vscod805cf:30301675
binariesv615:30325510
vsaa593:30376534
py29gd2263:30899288
vscaac:30438847
c4g48928:30535728
azure-dev_surveyone:30548225
a9j8j154:30646983
0bi6i642:30951795
pythongtdpath:30769146
welcomedialog:30910333
pythonidxpt:30866567
pythonnoceb:30805159
asynctok:30898717
pythontestfixt:30902429
pythonregdiag2:30936856
pyreplss1:30897532
pythonmypyd1:30879173
pythoncet0:30885854
pythontbext0:30879054
accentitlementsc:30887149
dsvsc016:30899300
dsvsc017:30899301
dsvsc018:30899302
dsvsc019bcf:30953938
7j2b6412:30949929

```

</details>


[Trace-20240206T004209.json](https://github.com/microsoft/vscode/files/14176540/Trace-20240206T004209.json)

**Comment by VSCodeTriageBot** [comment]
Thanks for creating this issue! We figured it's missing some basic information or in some other way doesn't follow our [issue reporting guidelines](https://aka.ms/vscodeissuereporting). Please take the time to review these and update the issue.

Happy Coding!

**Comment by jsecrest** [comment]
Seems to be helped by installing the previous version of pylance (2023.11.10)

**Comment by ErickPshat** [comment]
Same issue here. I find that this happens for example when I start typing the function like pd.to_datetime(), regardless of whether I am working in a notebook or a .py file. Maybe it has to do with pylance and parameter hints.
Downgrading to a previous VS Code version solves the issue. Downgrading to previous version of pylance does not.

**Comment by netbrothers-tr** [comment]
I can confirm this behaviour on Ubuntu Linux 22 since VS Code version 1.86.0.

**Comment by jrieken** [comment]
Please use this guide to provide more information: https://github.com/Microsoft/vscode/wiki/Performance-Issues. Thanks.

**Comment by RafaelFeiten** [comment]
Same issue here using two different computers.
When using the format functionality when saving the file, vscode hangs waiting for it to process, which takes a few seconds. The behavior does not always occur.

**Comment by jsecrest** [comment]
@jrieken you've referenced the performance gathering instructions a few times. I originally tried to follow them but the trace I downloaded from my recorded profile was empty. What is the correct way to save the profile? The instructions don't say.

**Comment by jrieken** [comment]
@jsecrest what precisely is the performance problem that you are experiencing. Slow startup? Slow editor features, like slow to complete or slow to show the outline? Slow, unresponsive UI like mouse moves stutter, typing in the editor and/or other input boxes is delayed. Depending on what you experience a different way to diagnose is needed. 


To everyone: we believe you but without exact information we cannot help you. So, refrain from "me too" comments, use the 👍-feature, but most importantly include detailed information (scenario, extensions, version, workload etc). It's also likely that same symptoms (slow/sluggish) aren't due to the same cause, so unless you have matching, detailed information file a new issue instead.

**Comment by Bakuretso** [comment]
> @jsecrest what precisely is the performance problem that you are experiencing. Slow startup? Slow editor features, like slow to complete or slow to show the outline? Slow, unresponsive UI like mouse moves stutter, typing in the editor and/or other input boxes is delayed. Depending on what you experience a different way to diagnose is needed.
> 
> To everyone: we believe you but without exact information we cannot help you. So, refrain from "me too" comments, use the 👍-feature, but most importantly include detailed information (scenario, extensions, version, workload etc). It's also likely that same symptoms (slow/sluggish) aren't due to the same cause, so unless you have matching, detailed information file a new issue instead.

Working with torch, matplotlib, sklearn in WSL.  When you type parameters (for example plt.figure(figsize= ) VS Code freezes for about 3-5 seconds. 

My PC spec is pretty decent r9 5950x, 64gb ram, nvme 980 pro for wsl. So, it can't be hardware issues. And there was no problem before I updated python extension to v2024.0.1 about 3 days ago.

**Comment by jrieken** [comment]
> VS Code freezes for about 3-5 seconds.

Describe "freeze" does the cursor stop blinking or do parameter help not update as you type more?

**Comment by chetanrchetu** [comment]
@jsecrest yes even me i am facing the same issue the freeze which @Bakuretso tells ,
like when i am working with maching learning related code like pandas or matplotlib in general machine learning stuff when the editor recommends the some inbulit function or tells someting about the function the system freezes even when accessing the variable name from the pandas dataframe,but this doesnt occur when i am dealing with flask or html only when i am doing maching learning code

**Comment by jrieken** [comment]
@chetanrchetu define "freeze", see https://github.com/microsoft/vscode/issues/204234#issuecomment-1934368095

**Comment by chetanrchetu** [comment]
> @chetanrchetu define "freeze", see [#204234 (comment)](https://github.com/microsoft/vscode/issues/204234#issuecomment-1934368095)



> @chetanrchetu define "freeze", see [#204234 (comment)](https://github.com/microsoft/vscode/issues/204234#issuecomment-1934368095)

Sir its kinda both cursor stops and even the help freezes

**Comment by jrieken** [comment]
If you can consistently cause the cursor to stop blinking that follow these steps to create CPU profile: https://github.com/Microsoft/vscode/wiki/Performance-Issues#profiling-the-renderer-process

**Comment by jsecrest** [comment]
I think I better understand what you're asking. I'm a little slammed at the
moment but I will try to get you a report with better details either
tonight or tomorrow. I will also use the performance tool to generate
another profile and if I still can't export it appropriately, I will at
least take a few screenshots and detail some of the information.

**Comment by netbrothers-tr** [comment]
After some more tests, I am convinced that - in my case - it has to do with one or more extensions and not with VS Code directly. _Extension Bisect_ unfortunately didn't work for me. I couldn't find the problem with it. But with all extensions disabled - in my case - the problem was gone. Now I will reactivate the extensions one by one to narrow it down and report back.

**Comment by jrieken** [comment]
Closing due to a lack of actionable information. The OP didn't get back with the requested information which means we cannot investigate/fix. Also bystanders also didn't provide information. 

**Important** I believe that you experiemence bad performance but please don't comment on an unconfirmed issue. Same sympton doesn't mean same cause. Please file separate issues, and please response do instructions for more information. That's the only sustainable way to improve things for you.

**Comment by jrieken** [comment]
@Mond45 Thanks a ton. The save-btn is that arrow-up icon (see below). Really hungry for the full profile now. Also wonder what UI element shows when you are seeing this? Does this happen as you type? Is this suggest with details or signature help


<img width="202" alt="Screenshot 2024-02-19 at 17 59 46" src="https://github.com/microsoft/vscode/assets/1794099/d6c91d33-b48d-452c-b672-8f9006a41efa">

**Comment by Mond45** [comment]
> > @jsecrest what precisely is the performance problem that you are experiencing. Slow startup? Slow editor features, like slow to complete or slow to show the outline? Slow, unresponsive UI like mouse moves stutter, typing in the editor and/or other input boxes is delayed. Depending on what you experience a different way to diagnose is needed.
> > To everyone: we believe you but without exact information we cannot help you. So, refrain from "me too" comments, use the 👍-feature, but most importantly include detailed information (scenario, extensions, version, workload etc). It's also likely that same symptoms (slow/sluggish) aren't due to the same cause, so unless you have matching, detailed information file a new issue instead.
> 
> Working with torch, matplotlib, sklearn in WSL. When you type parameters (for example plt.figure(figsize= ) VS Code freezes for about 3-5 seconds.
> 
> My PC spec is pretty decent r9 5950x, 64gb ram, nvme 980 pro for wsl. So, it can't be hardware issues. And there was no problem before I updated python extension to v2024.0.1 about 3 days ago.

It's the same as mentioned here, I'm using Python. The freezes happen when I'm using matplotlib, and sklearn.
![image](https://github.com/microsoft/vscode/assets/8156595/15ca12aa-d813-4f5b-8319-45a218c50948)
Can't seem to save the profile.

**Comment by jrieken** [comment]
@deepak1556 Can you please up on not being able to save profiles, see https://github.com/microsoft/vscode/issues/204234#issuecomment-1952899878

**Comment by Mond45** [comment]
@jrieken It's true that those libraries do have long documentations for some functions, still the whole renderer process should not be freezing. To reproduce this, you could try plt.figure function from matplotlib, and np.argmin from numpy.

**Comment by jrieken** [comment]
No success with neither of them... When rendering markdown we cap the input length to 100000 characters which is quite conservative. The sanitize call with those functions is <1ms for me, there must be a very specific case you are hidden, like one which doesn't go well with some sanitze regexes we use. Tho, I need your help and reliable repro case. Also sure what version of vscode and python/pylance you are using

**Comment by jsecrest** [comment]
I've been trying to reproduce my issue for about a week, but I've either
stopped having the issue, or I'm not able to spend enough time. In my case,
the issue would start up after an hour or two of working, and I've only had
time for shorter dev sessions this week.

To clarify my symptoms: the application would seem to freeze while typing
anything that might cause a linter popup for auto competition, especially
when getting the doc strings for a function. By "freeze" I mean that it
would stop accepting mouse and keyboard input. However, keyboard input was
still tracked, as after the freeze, the keyboard input I typed while frozen
would appear.

When I attempted to get profile information to share, I saw long jobs that
I think were in an animation category. I wish I had taken a screenshot. If
I'm able to spend enough time to recreate the issue I'll report back, but
for now I'm not able to replicate it.

Jason Secrest
